numba_cuda/numba/cuda/compiler.py

```diff
@@ -118,6 +118,10 @@ def run_pass(self, state):
         """
         lowered = state["cr"]
         signature = typing.signature(state.return_type, *state.args)
+        if "metadata" in state:
```

Comment:
Are there situations where `metadata` is not in `state`, or is this "just in case"?

Reply:
It's just in case, I reran the tests by accessing `.metadata` unconditionally and all passes.

Reply:
So I removed the check in [ff7852c](https://github.com/NVIDIA/numba-cuda/pull/227/commits/ff7852c87bbf3477c4ea4e5f3f5d5946a15ab92e)